src/main/java/com/iheartradio/m3u8/data/TrackData.java

```diff
@@ -94,6 +94,9 @@ public TrackData build() {
             if (mLocationType == null) {
                 throw new IllegalStateException("cannot build TrackData without a path or url");
             }
+            if (mTrackInfo == null) {
+                throw new IllegalStateException("cannot build TrackData without TrackInfo");
+            }
```

Comment:
Non-extended playlists do not have tags and thus can build tracks without track info.